pontoon/base/utils.py

```diff
@@ -606,3 +609,33 @@ def is_email(email):
         return True
     except ValidationError:
         return False
+
+
+def get_project_or_redirect(
+    slug, redirect_view_name, slug_arg_name, request_user, **kwargs
+):
+    """
+    Attempts to get a project with the given slug. If the project doesn't exist, it checks if the slug is in the
+    ProjectSlugHistory and if so, it redirects to the current project slug URL. If the old slug is not found in the
+    history, it raises an Http404 error.
+    """
+    Project = apps.get_model("base", "Project")
+    ProjectSlugHistory = apps.get_model("base", "ProjectSlugHistory")
+    try:
+        project = get_object_or_404(
+            Project.objects.visible_for(request_user).available(), slug=slug
+        )
```

Comment:
This wrapper provides no use, as the potential error is being immediately caught; see the example in the docs for what it's doing: https://docs.djangoproject.com/en/3.2/topics/http/shortcuts/#get-object-or-404